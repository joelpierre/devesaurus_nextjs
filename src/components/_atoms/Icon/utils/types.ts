import { ImgHTMLAttributes } from 'react';

export type TIconProps = ImgHTMLAttributes<HTMLImageElement> & {
  className?: string;
  inline?: boolean;
};

export enum EIconType {
  ChevronLeft = 'ChevronLeft',
  ChevronRight = 'ChevronRight',
  Logo = 'Logo',
  LogoDark = 'LogoDark',
  LogoCharacter = 'LogoCharacter',
  LogoCharacterDark = 'LogoCharacterDark',
  LogoSocial = 'LogoSocial',
  LogoSocialDark = 'LogoSocialDark',
  LogoSymbol = 'LogoSymbol',
  LogoSymbolDark = 'LogoSymbolDark',
  LogoText = 'LogoText',
  LogoTextDark = 'LogoTextDark'
}

export enum EIconTheme {
  Light = 'Light',
  Dark = 'Dark'
}
