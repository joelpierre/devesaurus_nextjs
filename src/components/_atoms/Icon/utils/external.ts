import { EIconType } from '@jpp/atoms/Icon/utils/types';

const IconUrls: Record<EIconType, string> = {
  [EIconType.ChevronLeft]: require('../../../../assets/svg/dev-chevron-left.svg'),
  [EIconType.ChevronRight]: require('../../../../assets/svg/dev-chevron-right.svg'),
  [EIconType.Logo]: require('../../../../assets/svg/logo.svg'),
  [EIconType.LogoDark]: require('../../../../assets/svg/logo-inv.svg'),
  [EIconType.LogoCharacter]: require('../../../../assets/svg/logo-character.svg'),
  [EIconType.LogoCharacterDark]: require('../../../../assets/svg/logo-character-inv.svg'),
  [EIconType.LogoSocial]: require('../../../../assets/svg/logo-social.svg'),
  [EIconType.LogoSocialDark]: require('../../../../assets/svg/logo-social-inv.svg'),
  [EIconType.LogoSymbol]: require('../../../../assets/svg/logo-symbol.svg'),
  [EIconType.LogoSymbolDark]: require('../../../../assets/svg/logo-symbol-inv.svg'),
  [EIconType.LogoText]: require('../../../../assets/svg/logo-text.svg'),
  [EIconType.LogoTextDark]: require('../../../../assets/svg/logo-text-inv.svg')
};

export default IconUrls;
