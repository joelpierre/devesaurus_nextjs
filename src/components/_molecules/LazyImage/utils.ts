export interface IRectangle {
  top: number;
  left: number;
  right: number;
  bottom: number;
}

export const isAllInViewport = (rect: IRectangle, viewportHeight: number, viewportWidth: number) => {
  return (
    rect.top >= 0 &&
    rect.left >= 0 &&
    rect.bottom <= viewportHeight &&
    rect.right <= viewportWidth
  );
};

export const isAnyInViewport = (rect: IRectangle, viewportHeight: number, viewportWidth: number) => {
  return (
    (rect.top >= 0 && rect.top <= viewportHeight ||
      rect.bottom >= 0 && rect.bottom <= viewportHeight) &&
    (rect.left >= 0 && rect.left <= viewportWidth ||
      rect.right >= 0 && rect.right <= viewportWidth)
  );
};

export const isElementInViewport = (element?: HTMLElement): boolean => {
  if (element === undefined
    || window === undefined
    || document === undefined
    || document.documentElement === null) {
    return false;
  }

  const rect = element.getBoundingClientRect();

  return isAllInViewport(
    rect,
    (window.innerHeight || document.documentElement.clientHeight),
    (window.innerWidth || document.documentElement.clientWidth)
  );
};

export const isElementNearViewport = (
  element: HTMLElement,
  portraitExpansionQuotient: number = 1,
  landscapeExpansionQuotient: number = 1): boolean => {

  if (element === undefined
    || window === undefined
    || document === undefined
    || document.documentElement === null) {
    return false;
  }

  const rect = element.getBoundingClientRect();

  let screenHeight = window.innerHeight || document.documentElement.clientHeight;
  let screenWidth = window.innerWidth || document.documentElement.clientWidth;

  const aspectRatio = screenWidth / screenHeight;

  const expansionQuotient = aspectRatio < 1
    ? portraitExpansionQuotient
    : landscapeExpansionQuotient;

  screenHeight += screenHeight * expansionQuotient;
  screenWidth += screenWidth * expansionQuotient;

  return isAnyInViewport(
    rect,
    screenHeight,
    screenWidth
  );
};

export const loadImageWithCanvas = (imageUrl: string): Promise<string> => {
  return new Promise<string>((resolve, reject) => {

    return loadImage(imageUrl).then(onLoadImage);

    // tslint:disable-next-line:only-arrow-functions
    function onLoadImage(image: HTMLImageElement) {
      const canvas = createImageCanvas(image);

      const context = canvas.getContext('2d');

      if (context !== null) {
        context.drawImage(image, 0, 0);
        const dataUrl = canvas.toDataURL('image/png');
        resolve(dataUrl);
      } else {
        reject('Unable to get 2d canvas.');
      }
    }
  });
};

export const loadImageWithXhr = (imageUrl: string): Promise<string> => {
  return new Promise<string>((resolve, reject) => {
    try {
      const xhr = new XMLHttpRequest();
      const blobHandler = () => {
        const url = window.URL.createObjectURL(xhr.response);
        resolve(url);
      };

      const svgHandler = () => {
        const svgData = encodeURIComponent(xhr.response);
        const url = `"data:image/svg+xml; charset=utf8, ${svgData}"`;
        resolve(url);
      };

      if (imageUrl.endsWith('.svg')) {
        xhr.onload = svgHandler;
      } else {
        xhr.onload = blobHandler;
        xhr.responseType = 'blob';
      }

      xhr.open('GET', imageUrl, true);
      xhr.send();

    } catch (error) {
      reject(error);
    }
  });
};

export const isFullBleed = (image: Core.IMedia) => {
  const getImageProp = (key: keyof Core.IMedia, defaultValue: number): number => {
    return image && image[key] as number || defaultValue;
  };

  const width = getImageProp('width', 0);
  const height = getImageProp('height', 0);
  const diff = Math.abs(width - height);

  // DIFF_TOLERANCE is an arbitrary about the diff is allowed to vary by.
  const DIFF_TOLERANCE = 5;
  return diff >= DIFF_TOLERANCE && width > 1000;
};

export const loadImage = (src: string): Promise<HTMLImageElement> => {
  return new Promise((resolve, reject) => {
    try {
      const image = new Image();
      image.crossOrigin = 'anonymous';
      image.src = src;
      image.onload = () => resolve(image);
      image.onerror = reject;
    } catch (error) {
      reject(error);
    }
  });
};

export const createImageCanvas = (image: HTMLImageElement): HTMLCanvasElement => {
  const canvas = document.createElement('canvas');
  const { naturalWidth, naturalHeight } = image;
  canvas.width = naturalWidth;
  canvas.height = naturalHeight;
  return canvas;
};
