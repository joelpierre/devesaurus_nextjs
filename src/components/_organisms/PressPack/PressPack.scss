.press-pack {
  $root: &;
  position: relative;

  &__button {
    margin-top: 3rem;
    margin-bottom: 3rem;
  }

  &__image-wrapper {
    overflow: hidden;
  }

  &__image {
    width: 100%;
    max-width: 100% !important;
    height: auto;
    max-height: 300px !important;
  }
}
