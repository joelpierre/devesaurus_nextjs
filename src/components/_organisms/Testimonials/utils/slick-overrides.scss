.testimonials-slider {
  .slick-list {
    overflow: hidden;
  }

  .slick-track {
    display: flex;
    align-items: center;
  }
}
