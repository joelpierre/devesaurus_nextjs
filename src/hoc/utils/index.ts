import { ComponentType } from 'react';

export const getDisplayName = <T extends object>(
  WrappedComponent: ComponentType<T>,
) => {
  return WrappedComponent.displayName || WrappedComponent.name || 'Component';
};
