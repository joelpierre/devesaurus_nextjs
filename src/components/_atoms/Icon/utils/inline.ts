import { FunctionComponent } from 'react';

import { EIconType, TIconProps } from '@jpp/atoms/Icon/utils/types';

const IconComponents: Record<EIconType, FunctionComponent<TIconProps>> = {
  [EIconType.ChevronLeft]: require('-!svg-react-loader?props[]=class:svg!../../../../assets/svg/dev-chevron-left.svg'),
  [EIconType.ChevronRight]: require('-!svg-react-loader?props[]=class:svg!../../../../assets/svg/dev-chevron-right.svg'),
  [EIconType.Logo]: require('-!svg-react-loader?props[]=class:svg!../../../../assets/svg/logo.svg'),
  [EIconType.LogoDark]: require('-!svg-react-loader?props[]=class:svg!../../../../assets/svg/logo-inv.svg'),
  [EIconType.LogoCharacter]: require('-!svg-react-loader?props[]=class:svg!../../../../assets/svg/logo-character.svg'),
  [EIconType.LogoCharacterDark]: require('-!svg-react-loader?props[]=class:svg!../../../../assets/svg/logo-character-inv.svg'),
  [EIconType.LogoSocial]: require('-!svg-react-loader?props[]=class:svg!../../../../assets/svg/logo-social.svg'),
  [EIconType.LogoSocialDark]: require('-!svg-react-loader?props[]=class:svg!../../../../assets/svg/logo-social-inv.svg'),
  [EIconType.LogoSymbol]: require('-!svg-react-loader?props[]=class:svg!../../../../assets/svg/logo-symbol.svg'),
  [EIconType.LogoSymbolDark]: require('-!svg-react-loader?props[]=class:svg!../../../../assets/svg/logo-symbol-inv.svg'),
  [EIconType.LogoText]: require('-!svg-react-loader?props[]=class:svg!../../../../assets/svg/logo-text.svg'),
  [EIconType.LogoTextDark]: require('-!svg-react-loader?props[]=class:svg!../../../../assets/svg/logo-text-inv.svg'),
};

export default IconComponents;
