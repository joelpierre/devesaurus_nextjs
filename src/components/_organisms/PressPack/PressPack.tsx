import React, { FunctionComponent } from 'react';
import classNames from 'classnames';
import Container from 'src/components/grid/Container/Container';
import Flex from 'src/components/grid/Flex/Flex';
import Row from 'src/components/grid/Row/Row';
import Section from 'src/components/grid/Section/Section';
import Button from 'src/components/molecules/Buttons/Button';
import Heading from 'src/components/shared/Heading/Heading';
import { IAcfModule } from 'src/utils/interfaces';
import { TTheme } from 'src/utils/types';

import styles from './PressPack.module.scss';

interface IPressPackProps {
  className: string;
  pageTheme: TTheme;
  module: Partial<IAcfModule>;
}

const PressPack: FunctionComponent<Partial<IPressPackProps>> = ({
  module = {},
  pageTheme = 'tint-alpha',
  className,
}) => {
  const { heading, copy, brand_zip_file, image, theme } = module;
  const {
    url: { source_url: ZIP_URL },
  } = brand_zip_file;
  const {
    source_url,
    media_details: { width, height },
  } = image;

  const imageAttrs = {
    loading: 'lazy'
  };

  return (
    <Section
      data-test="component-press-pack"
      theme={theme ? theme : pageTheme}
      className={classNames(styles.pressPack, className)}
    >
      <Container fluid={false}>
        <Row>
          <Flex colLg={6} className={styles.pressPackContent}>
            <Heading className={styles.pressPackHeading}>
              {heading ? heading : 'Press Pack'}
            </Heading>
            <p
              className={styles.pressPackCopy}
              dangerouslySetInnerHTML={{ __html: copy }}
            />
            <Button
              className={styles.pressPackButton}
              theme={'brand'}
              behaviour={'anchor'}
              link={ZIP_URL}
            >
              Download our Press Pack
            </Button>
          </Flex>

          <Flex
            colLg={6}
            className={classNames(styles.pressPackImageWrapper, 'ml-auto')}
          >
            <img
              src={source_url}
              className={styles.pressPackImage}
              width={width}
              height={height}
              alt="Press pack supportive image"
              {...imageAttrs}
            />
          </Flex>
        </Row>
      </Container>
    </Section>
  );
};

export default PressPack;
