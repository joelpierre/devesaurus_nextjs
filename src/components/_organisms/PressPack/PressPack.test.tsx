import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import { findByTestAttr, matchSnapshot } from '../../../utils/test';

import PressPack from './PressPack';

const defaultProps = {
  module: {
    brand_zip_file: {
      wordpress_id: 588,
      url: {
        wordpress_id: 588,
        source_url:
          'http://pixeltwist.test/wp-content/uploads/2019/08/Logo.zip',
      },
    },
    image: {
      wordpress_id: 580,
      alt_text: '',
      id: 'caca0077-86c2-55cf-9f48-37de2a73dd22',
      source_url: '/08/459366921_medium.jpg',
      media_type: 'image',
      media_details: {
        width: 2227,
        height: 1348,
      },
    },
  },
};

/**
 * Factory function to create a ShallowWrapper for the PressPack component.
 * @param {object} props - Component props specific to setup
 * @returns {ShallowWrapper}
 */
const setup = (props = {}) => {
  const setupProps = { ...defaultProps, ...props };
  return shallow(<PressPack {...setupProps} />);
};

describe('<PressPack/>', () => {
  let wrapper: ShallowWrapper;

  beforeEach(() => {
    wrapper = setup();
  });

  it('renders the PressPack Component without errors', () => {
    const component = findByTestAttr(wrapper, 'component-press-pack');
    expect(component.length).toBe(1);
  });

  it('should match snapshot', () => {
    matchSnapshot(wrapper);
  });
});
