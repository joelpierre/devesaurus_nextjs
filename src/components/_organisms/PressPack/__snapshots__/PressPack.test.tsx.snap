// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PressPack/> should match snapshot 1`] = `
<Section
  className="pressPack"
  data-test="component-press-pack"
  theme="tint-alpha"
>
  <Container
    fluid={false}
  >
    <Row>
      <Flex
        className="pressPackContent"
        colLg={6}
      >
        <Heading
          className="pressPackHeading"
        >
          Press Pack
        </Heading>
        <p
          className="pressPackCopy"
          dangerouslySetInnerHTML={
            Object {
              "__html": undefined,
            }
          }
        />
        <Button
          behaviour="anchor"
          className="pressPackButton"
          link="http://pixeltwist.test/wp-content/uploads/2019/08/Logo.zip"
          theme="brand"
        >
          Download our Press Pack
        </Button>
      </Flex>
      <Flex
        className="pressPackImageWrapper ml-auto"
        colLg={6}
      >
        <img
          alt="Press pack supportive image"
          className="pressPackImage"
          height={1348}
          loading="lazy"
          src="/08/459366921_medium.jpg"
          width={2227}
        />
      </Flex>
    </Row>
  </Container>
</Section>
`;
